test/geolocation_service_test.dart

```diff
@@ -0,0 +1,38 @@
+import 'package:flutter/widgets.dart';
+import 'package:location/location.dart';
+import 'package:mockito/mockito.dart';
+import 'package:test/test.dart';
+import 'package:veggie_go_malaysia/services/geolocation_service.dart';
+
+class MockLocation extends Mock implements Location {}
+
+void main() {
+  WidgetsFlutterBinding.ensureInitialized();
+  final geolocationService = GeoLocationService();
+  group('GeoLocationService', () {
+    test(
+        'returns a location if the service is enabled and permission is granted completes successfully',
```

Comment:
this isn't really testing anything then since everything is being overridden in your test to give you the output that you've hardcoded it to give... there is really no benefit to this afaik

Reply:
Everything is being overridden because I don't want the call to call the actual service itself, I am mocking what gets returned when specific calls are made.
Let me know if you have a better way? I'm new to this

Reply:
I followed the flutter doc to do the testing: https://flutter.dev/docs/cookbook/testing/unit/mocking